server/sonar-server/src/main/java/org/sonar/server/es/IndexingListener.java

```diff
@@ -19,13 +19,23 @@
  */
 package org.sonar.server.es;
 
-import java.util.Collection;
+import java.util.List;
 
 public interface IndexingListener {
 
-  void onSuccess(Collection<String> docIds);
+  void onSuccess(List<DocId> docIds);
 
-  static IndexingListener noop() {
-    return docIds -> {};
-  }
+  void onFinish(IndexingResult result);
+
+  IndexingListener NOOP = new IndexingListener() {
+    @Override
+    public void onSuccess(List<DocId> docIds) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1186)